### DISCUSSION TITLE
Pi-Alerts

### DISCUSSION THREAD

**Discussion by Ma3er** [tteck/Proxmox#1292] [Ideas]
Are there any plans to add Pi-Alerts? `https://github.com/jokob-sk/Pi.Alert`

**Comment by killtux** [comment]
i would appreciate this as well

**Comment by jokob-sk** [comment]
Not sure if still interested, but I added an on-HW install script: https://github.com/jokob-sk/Pi.Alert/blob/main/docs/HW_INSTALL.md

**Reply by mgrimace** [reply]
Can the script detect an existing installation of Pi-Hole and automatically get Pi-Hole devices imported? 

Could this script be included as optional add-on to the existing Pi-Hole LXC script? For example, to have both Pi-Hole and Pi.Alert installed on the same LXC?

I'm new to Pi.Alert and it seems that it can make use of [Pi-Hole data ](https://github.com/jokob-sk/Pi.Alert/blob/main/dockerfiles/README.md#for-pihole-pihole_run-dhcplss_run)

**Reply by jokob-sk** [reply]
In theory, it's possible, but I don't have Proxmox setup on my end to test this. I would have to rely on a bit of community support to get this implemented and tested. Happy to collaborate and get this working together, if interested. 

To get PiAlert and PiHole working together from the beginning only 3 settings need to be set (this is for the DB mapping approach):

  - `PIHOLE_RUN` is used to enable the import by setting it e.g. to `schedule` or `once`   (pre-set to `disabled`)
  - `PIHOLE_RUN_SCHD` is to configure how often the plugin is executed if `PIHOLE_RUN` is set to `schedule` (pre-set to every 30 min)
  - `PIHOLE_DB_PATH` setting must match the location of your PiHole database (pre-set to `/etc/pihole/pihole-FTL.db`)

**Reply by mgrimace** [reply]
Thank you for the reply, I'm definitely interested. In essence it shouldn't matter that it's Proxmox per se, it should function like any Debian server (e.g., VM, etc.). I'm happy to test, but I can't contribute to the coding side of things.

In terms of those 3 settings - where are those entered (e.g., after your install script is run, assuming it's on the same machine as the Pi-Hole?). Thanks again!
~M

**Reply by jokob-sk** [reply]
They are saved/read from this file:

https://github.com/jokob-sk/Pi.Alert/blob/main/config/pialert.conf

So having a variant of this file for a PiHole Proxmox setup might be feasible. Here is an example of how these settings are defined:

```
PIHOLE_RUN='schedule'
PIHOLE_DB_PATH='/etc/pihole/pihole-FTL.db'
PIHOLE_RUN_SCHD='*/30 * * * *'
```
I however have no idea what these values (especially for `PIHOLE_DB_PATH` should default to on a Proxmox setup and how to decide if you want to enable the default PiHole integration with PiAlert via a proxmox config (if there is such a thing?).

**Reply by mgrimace** [reply]
AFAIK the TTeck script in this discussion just makes a standard Debian linux container with 512 ram, 1vCPU, and 2gb storage and automates the default installtion of Pi-Hole on the system. I don't believe there's any special config required - Proxmox is just the hypervisor/server manager that allows you to spin up various linux containers (LXCs) and VMs, and TTeck makes scripts to automate the creation of such containers with pre-installed software.

I ran your Pi.Alert script on my existing LXC with Pi-Hole and received the following errors using both curl and wget:
```bash

Download Finished

Updating... /usr/share/arp-scan/
/home/pi/pialert/back/update_vendors.sh: 45: cd: can't cd to /usr/share/arp-scan
sudo: get-iab: command not found
sudo: get-oui: command not found
chmod: cannot access '/home/pi/front/log': No such file or directory
chmod: cannot access '/home/pi/config': No such file or directory
Starting php7.4-fpm (via systemctl): php7.4-fpm.service.
./install.sh: line 81: /etc/init.d/nginx: No such file or directory
Failed to start nginx.service: Unit nginx.service not found.
Failed to enable unit: Unit file nginx.service does not exist.
Failed to enable unit: Unit file pi-alert.service does not exist.
Failed to restart nginx.service: Unit nginx.service not found.
Installation completed. Please configure any additional settings for your application.
Nothing to do here.
Starting php7.4-fpm (via systemctl): php7.4-fpm.service.
/home/pi/pialert/dockerfiles/start.sh: 28: /etc/init.d/nginx: not found
Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/pi/pialert/pialert/__main__.py", line 27, in <module>
    from helper import  filePermissions, timeNowTZ, updateState, get_setting_value, noti_obj
  File "/home/pi/pialert/pialert/helper.py", line 10, in <module>
    import pytz
ModuleNotFoundError: No module named 'pytz'

```
For reference it is Debian 12.2.0-14 and has Pi-Hole in the default locations (e.g., /etc/pihole/pihole-FTL.db as above)
Should I bring that over into your community discussion?

**Reply by tteck** [reply]
There are several reasons why this won't function as is without making modifications to the installation script. Issues may include incorrect paths and the use of outdated PHP/Python versions, among others.

**Reply by mgrimace** [reply]
Understandable, it seems like the best use of Pi.Alerts is to have it on the same system as a Pi-Hole, I'll tinker around with that on my end, appreciate all the good work!

**Reply by tteck** [reply]
To add Pi.Alert to an existing Pi-hole LXC, execute
```
bash -c "$(wget -qLO - https://github.com/leiweibau/Pi.Alert/raw/main/install/pialert_install.sh)"
```
in the Pi-hole LXC console and follow the instructions.

**Reply by jokob-sk** [reply]
@mgrimace I rewrote the installation scripts so you can try running the below command again. I tested it on an empty `bookworm-slim` container.

```bash
curl -o install.sh https://raw.githubusercontent.com/jokob-sk/Pi.Alert/main/install/install.sh && sudo chmod +x install.sh && sudo ./install.sh
```

**Reply by mgrimace** [reply]
Using the leiweibau script above per Tteck worked as expected - up and running.

I also tested the updated jakob-sk script on a fresh Pi-Hole install, and it appears to be repeating the following loop:

```bash
16:51:38 [Send API] Updating notification_* files in /home/pi/pialert/front/api/
16:51:38 [Notification] No changes to report
16:51:38 [MAIN] Process: Wait
16:51:43 [MAIN] waiting to start next loop
16:51:49 [MAIN] waiting to start next loop
16:51:54 [MAIN] waiting to start next loop
16:51:59 [MAIN] waiting to start next loop
16:52:04 [MAIN] waiting to start next loop
16:52:09 [MAIN] waiting to start next loop
16:52:14 [MAIN] waiting to start next loop
16:52:19 [MAIN] waiting to start next loop
16:52:24 [MAIN] waiting to start next loop
16:52:29 [MAIN] waiting to start next loop
16:52:34 [MAIN] waiting to start next loop
16:52:39 [Notification] Check if something to report
16:52:39 [Notification] Open text Template
16:52:39 [Notification] Open html Template
16:52:39 [Notification] Using template/back/report_template.html
16:52:39 [Notification] included sections: ['internet', 'new_devices', 'down_devices', 'events']
16:52:39 [Notification] New Devices sections done.
16:52:39 [Notification] Down Devices sections done.
16:52:39 [Notification] Events sections done.
16:52:39 [Notification] Udating API files
```

**Reply by tteck** [reply]
The branch by @leiweibau is geared more toward bare metal environments, while the branch by @jokob-sk is oriented towards Docker environments.

**Reply by jokob-sk** [reply]
@mgrimace  That is expected, the app is running and should be accessible on the port 20211 👍

**Comment by Pinez83** [comment]
> Not sure if still interested, but I added an on-HW install script: https://github.com/jokob-sk/Pi.Alert/blob/main/docs/HW_INSTALL.md

Great! Can this be used to install Pi-Alerts on a Vm/lxc on proxmox? Do I just need to copy this string into the proxmox shell or do I have to first create the vm/lxc with the appropriate operating system?

**Reply by tteck** [reply]
There is a **Pi.Alert LXC** using the [leiweibau](https://github.com/leiweibau/Pi.Alert/) branch in this repo.

**Reply by tteck** [reply]
If for some reason you want the [jokob-sk](https://github.com/jokob-sk/Pi.Alert) branch of Pi.Alert also, let me know.

**Reply by Pinez83** [reply]
I just used your script.
Thank you very much

**Reply by lylydang** [reply]
Hi, sorry for digging through this old topic and sorry for asking this, by any chance you have the update script for Pi Alert's Jokob version? The current one seems to be no longer maintained. Appreciate all your work!

**Reply by ggee** [reply]
Yes, the option for jokob-sk would be nice.

**Comment by rcv11x** [comment]
If I already have a LXC with pihole and now I want to install pi.alert it is better to install it through the new fork of "leiweibau" or I can run your script: `bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/pialert.sh)"` in the same LXC? or this script is only to create a LXC from 0, I have this doubt.

**Reply by tteck** [reply]
`bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/pialert.sh)"` creates a NEW LXC only.

**Comment by killtux** [comment]
> If for some reason you want the [jokob-sk](https://github.com/jokob-sk/Pi.Alert) branch of Pi.Alert also, let me know.

i like the fork of [jokob-sk](https://github.com/jokob-sk/Pi.Alert) more because its features. if you can add this too it would be awesome. anyway - is there a way to update the lxc if there is a new release?

**Reply by vikdb** [reply]
Its now called NetAlertX

**Comment by xHuubx** [comment]
Will there also be a LXC installer for [NetAlertX](https://github.com/jokob-sk/NetAlertX?tab=readme-ov-file) or maybe a update script? Would be great!
